src/librustc/ty/sty.rs

```diff
@@ -150,6 +150,9 @@ pub enum TypeVariants<'tcx> {
     /// `|a| a`.
     TyClosure(DefId, ClosureSubsts<'tcx>),
 
+    /// The anonymous type of a generator. Pairs with a TyClosure for closure generators.
```

Comment:
I think this comment could be expanded. For example, what does "pairs with" mean? My guess is that this is the type assigned to an expression like `|| yield 22`, right? In other words, it represents a "resumable" function that, each time it is called, picks up from the last yield point? (Only it works through the generator trait, not the function traits?)